oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/annotation/Column.java

```diff
@@ -87,6 +86,27 @@
      */
     ValueDataType dataType() default ValueDataType.NOT_VALUE;
 
+    /**
+     * The storage analyzer mode.
+     *
+     * @since 8.4.0
+     */
+    AnalyzerType analyzer() default AnalyzerType.OAP_ANALYZER;
+
+    /**
+     * The analyzer declares the text analysis mode.
+     */
+    enum AnalyzerType {
```

Comment:
As an enum type you should put more information in the type, rather than separating them in the codes.

Reply:
fixed

Reply:
You should add `getSettings` method to return the settings.

Reply:
I would add a functional interface to refactor it.

Reply:
As an enum, I think a method is enough. We would use the interface to anywhere else.